File: 2020/day1/day1.py
S = set(map(int, open("day1.txt")))
r1 = [(a, 2020 - a) for a in S if (2020 - a) in S][0]
r2 = [(a, b, 2020 - a - b) for a in S for b in S
      if a != b and (2020 - a - b) in S][0]
print(r1[0] * r1[1])
print(r2[0] * r2[1] * r2[2])

File: 2020/day1/day1.kt
import java.io.File

fun main() {
    val S = File("day1.txt").readLines()
        .map { it.toInt() }.toSet()
    val r1 = S.filter { (2020 - it) in S }.first()
    val r2 = S.flatMap { a -> S.mapNotNull { b ->
        if ((2020 - a - b) in S) Pair(a, b) else null } }.first()
    println(r1 * (2020 - r1))
    println(r2.first * r2.second * (2020 - r2.toList().sum()))
}
